### ISSUE TITLE
run fails on installation -- cannot find `sqlparse`

### ISSUE THREAD

**Issue by jkornblum** [conda-forge/dbt-feedstock#27]
Issue:
 When I install I get this error when trying to run
It shows dbt-core installed from pypi_0 but its not installing sqlparse which is `requires` in that projects `setup.py`
```
Traceback (most recent call last):
  File "/home/jk88/miniconda3/envs/emedb/bin/dbt", line 8, in <module>
    from dbt.main import main
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/main.py", line 12, in <module>
    import dbt.task.run as run_task
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/task/run.py", line 8, in <module>
    from dbt.node_runners import ModelRunner, RPCExecuteRunner
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/node_runners.py", line 5, in <module>
    from dbt.contracts.results import RunModelResult, collect_timing_info, \
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/contracts/results.py", line 4, in <module>
    from dbt.contracts.graph.manifest import COMPILE_RESULT_NODE_CONTRACT
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/contracts/graph/manifest.py", line 5, in <module>
    from dbt.contracts.graph.compiled import COMPILED_NODE_CONTRACT, CompiledNode
  File "/home/jk88/miniconda3/envs/emedb/lib/python3.7/site-packages/dbt/contracts/graph/compiled.py", line 8, in <module>
    import sqlparse
ModuleNotFoundError: No module named 'sqlparse'
```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

_libgcc_mutex             0.1                        main  
alembic                   1.2.1                      py_0    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
asn1crypto                0.24.0                   pypi_0    pypi
astroid                   2.3.0                    py37_0    conda-forge
attrs                     19.1.0                     py_0    conda-forge
awscli                    1.16.247                 py37_0    conda-forge
azure-common              1.1.23                   pypi_0    pypi
azure-storage-blob        2.1.0                    pypi_0    pypi
azure-storage-common      2.1.0                    pypi_0    pypi
backcall                  0.1.0                      py_0    conda-forge
black                     19.3b0                     py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
boto3                     1.9.233                  pypi_0    pypi
botocore                  1.12.233                 pypi_0    pypi
bzip2                     1.0.8                h516909a_1    conda-forge
ca-certificates           2019.9.11            hecc5488_0    conda-forge
certifi                   2019.9.11                py37_0    conda-forge
cffi                      1.12.3           py37h8022711_0    conda-forge
chardet                   3.0.4                    pypi_0    pypi
click                     7.0                        py_0    conda-forge
colorama                  0.3.9                      py_1    conda-forge
cryptography              2.7              py37h72c5cf5_0    conda-forge
dbt                       0.14.2                     py_0    conda-forge
dbt-bigquery              0.14.2                   pypi_0    pypi
dbt-core                  0.14.2                   pypi_0    pypi
dbt-postgres              0.14.2                   pypi_0    pypi
dbt-redshift              0.14.2                   pypi_0    pypi
dbt-snowflake             0.14.2                   pypi_0    pypi
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
docutils                  0.15.2                   py37_0    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
future                    0.17.1                   pypi_0    pypi
idna                      2.8                      pypi_0    pypi
ijson                     2.5                      pypi_0    pypi
ipykernel                 5.1.2            py37h5ca1d4c_0    conda-forge
ipython                   7.8.0            py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isort                     4.3.21                   py37_0    conda-forge
jedi                      0.15.1                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
jmespath                  0.9.4                      py_0    conda-forge
json5                     0.8.5                      py_0    conda-forge
jsonschema                3.0.2                    py37_0    conda-forge
jupyter_client            5.3.3                    py37_1    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterlab                1.1.4                      py_0    conda-forge
jupyterlab_server         1.0.6                      py_0    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
lazy-object-proxy         1.4.2            py37h516909a_0    conda-forge
libblas                   3.8.0               12_openblas    conda-forge
libcblas                  3.8.0               12_openblas    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libiconv                  1.15              h516909a_1005    conda-forge
liblapack                 3.8.0               12_openblas    conda-forge
libopenblas               0.3.7                h6e990d7_1    conda-forge
libpq                     11.5                 hd9ab2ff_1    conda-forge
libsodium                 1.0.17               h516909a_0    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
mako                      1.1.0                      py_0    conda-forge
markupsafe                1.1.1            py37h14c3975_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
mistune                   0.8.4           py37h14c3975_1000    conda-forge
nbconvert                 5.6.0                    py37_1    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
notebook                  6.0.1                    py37_0    conda-forge
numpy                     1.17.2           py37h95a1406_0    conda-forge
openssl                   1.1.1c               h516909a_0    conda-forge
pandas                    0.24.2           py37hb3f55d8_0    conda-forge
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.5.1                      py_0    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.2.3                   py37_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
psycopg2                  2.7.6.1          py37h1ba5d50_0  
ptyprocess                0.6.0                   py_1001    conda-forge
pyasn1                    0.4.6                      py_0    conda-forge
pycparser                 2.19                     pypi_0    pypi
pycryptodomex             3.9.0                    pypi_0    pypi
pygments                  2.4.2                      py_0    conda-forge
pyjwt                     1.7.1                    pypi_0    pypi
pylint                    2.3.1                    py37_0    conda-forge
pyodbc                    4.0.27           py37he1b5a44_0    conda-forge
pyopenssl                 19.0.0                   pypi_0    pypi
pyrsistent                0.15.4           py37h516909a_0    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
python-editor             1.0.4                      py_0    conda-forge
pytz                      2019.2                     py_0    conda-forge
pyyaml                    5.1.2            py37h516909a_0    conda-forge
pyzmq                     18.1.0           py37h1768529_0    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
requests                  2.22.0                   pypi_0    pypi
rsa                       3.4.2                      py_1    conda-forge
s3transfer                0.2.1                    py37_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.2.0                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
snowflake-connector-python 1.9.0                    pypi_0    pypi
snowflake-sqlalchemy      1.1.14                   pypi_0    pypi
sqlalchemy                1.3.8            py37h516909a_0    conda-forge
sqlite                    3.29.0               hcee41ef_1    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                   py_1001    conda-forge
tk                        8.6.9             hed695b0_1003    conda-forge
toml                      0.10.0                     py_0    conda-forge
tornado                   6.0.3            py37h516909a_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
unixodbc                  2.3.7             h227dcee_1000    conda-forge
urllib3                   1.25.5                   pypi_0    pypi
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.6                   py37_0    conda-forge
wrapt                     1.11.2           py37h516909a_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.1.7             h14c3975_1001    conda-forge
zeromq                    4.3.2                he1b5a44_2    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : emedb
    active env location : /home/jk88/miniconda3/envs/emedb
            shell level : 2
       user config file : /home/jk88/.condarc
 populated config files : 
          conda version : 4.7.12
    conda-build version : not installed
         python version : 3.6.7.final.0
       virtual packages : 
       base environment : /home/jk88/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/jk88/miniconda3/pkgs
                          /home/jk88/.conda/pkgs
       envs directories : /home/jk88/miniconda3/envs
                          /home/jk88/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.6.7 Linux/5.0.0-29-generic ubuntu/18.04.3 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</details>

**Comment by drewbanin** [comment]
Hey @jkornblum! conda-forge sends us automated PRs to release dbt through conda and I always merge them, but I was never sure that it actually worked!

Are you able to try installing via `pip` instead? I can definitely check this out (it would be great to have the conda distribution working) but I'm unsure how soon I'll be able to get to it!

**Comment by jkornblum** [comment]
I was able to install via pip successfully so I'm up and running.

I'm not an expert with conda build system but I've released 3 or 4 conda-forge feedstocks on here that appear to work :laughing: .
I'll add it to my TODO list to take a look at this repo and submit a PR.

--- FYI here are my initial thoughts, someone who knows more about conda/dbt may have different answer that would be less work. The conda-forge gitter channel is extremely responsive if you need assistance.

I think the "proper" way to create the feedstock recipe would be to have one conda-forge package for each dbt pypi package, in your case each of the `dbt-*` packages. Inside each recipe in the `requirements.run` section you would list the equivalent to the `install_requires` of the `setup.py`. For conda to properly (entirely) solve the dependency tree, each dependency of `dbt-*` would also need to be a conda-forge package and listed in the `run` section, for most common packages someone else has already completed this.

In the current recipe
- you have `--no-deps` specified so I don't _think_ pips going into the `dbt-*` dependency tree and installing those. --> this causes pip to not work correctly and why my run failed
- if you install all the `dbt-*` via `pip install` then conda won't be able to track them completely, i.e. they'll just be installed by pip (via conda cli) instead of through a conda channel. --> this will cause conda to not understand full dependency tree for an environment

So e.g. you'd have a feedstock recipe for `dbt-core` where the `requirements.run` section would look something like

```
# ...

requirements:
  host:
    - pip
    - python
    - setuptools
  run:
    - Jinja2>=2.10,
    - PyYAML>=3.11,
    - sqlparse=0.2.3,
    - networkx>=1.11,<3
    # REST OF DEPS

# ...
```
and then your dbt feedstock recipe (this repo) would look something like
```
# ...

requirements:
  host:
    - pip
    - python
    - setuptools
  run:
      - dbt-core=={{ version }}
      - dbt-postgres=={{ version }}
      - dbt-redshift=={{ version }}
      - dbt-snowflake=={{ version }}
      - dbt-bigquery=={{ version }}

# ...
```

here is example of [pysafety](https://github.com/conda-forge/safety-feedstock/blob/master/recipe/meta.yaml) package I got working on conda-froge

**Comment by thewchan** [comment]
@conda-forge/core @jkornblum  Okay, I'm in the process of building all individual dbt plugins as separate conda forge packages, this might take awhile because several dependencies also needed to be created within CF.

**Comment by maresb** [comment]
This should now be fixed by #60. In case it's not, ping me and I'll reopen.
